Integrations/integration-GoogleResourceManager.yml

```diff
@@ -0,0 +1,1020 @@
+        "client_email": CLIENT_EMAIL,
+        "client_id": CLIENT_ID,
+        "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+        "token_uri": "https://oauth2.googleapis.com/token",
+        "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+        "client_x509_cert_url": CLIENT_X509_CERT_URL
+    }
+
+    # Params for constructing googleapiclient service object
+    API_VERSION = 'v1'
+    GRM = 'cloudresourcemanager'
+    SCOPE = ["https://www.googleapis.com/auth/cloud-platform"]
+
+
+    ''' HELPER FUNCTIONS '''
+
+
+    def build_and_authorize():
+        """
+        Return a service object via which can call GRM API.
+
+        Use the service_account credential file generated in the Google Cloud
+        Platform to build the Google Resource Manager API Service object.
+
+        returns: service
+            Google Resource Manager API Service object via which commands in the
+            integration will make API calls
+        """
+        auth_json_string = str(AUTH_JSON).replace("\'", "\"").replace("\\\\", "\\")
+        service_account_info = json.loads(auth_json_string)
+        service_credentials = service_account.Credentials.from_service_account_info(service_account_info, scopes=SCOPE)
+        service = discovery.build(GRM, API_VERSION, credentials=service_credentials)
+        return service
+
+
+    def make_project_body(project_body):
+        """
+        Create and return the project body argument used when calling the GRM API
+        to create or update a project.
+
+        returns: (dict) body
+            dict object formatted to be used in the create or update API call
+        """
+        keys = project_body.keys()
+        body = {}
+        if 'projectId' in keys:
+            body['projectId'] = project_body['projectId']
+        if 'parent_type' in keys and 'parent_id' in keys:
+            body['parent'] = {
+                'type': project_body['parent_type'],
+                'id': project_body['parent_id']
+            }
+        if 'name' in keys:
+            body['name'] = project_body['name']
+        if 'labels.attrs' in keys and 'labels.values' in keys:
+            labels_attrs = argToList(project_body['labels.attrs'])
+            labels_values = argToList(project_body['labels.values'])
+            if len(labels_attrs) != len(labels_values):
+                err_msg = 'Label attrs array and Label values array do not match'\
+                    ' in length.\nThese arrays need to match in length because '\
+                    'each string in labels.attrs is assigned the value of the '\
+                    'string in the corresponding index in the labels.values array.'
+                raise IndexError(err_msg)
+            else:
+                body['labels'] = {}
+                for lbl_tag, lbl_val in zip(labels_attrs, labels_values):
+                    body['labels'][lbl_tag] = lbl_val
+        return body
+
+
+    def poll_operation(operation):
```

Comment:
we might want to use our `polling` mechanism here, how long does it take for this to run ?